packages/dev-env/src/introspect.ts

```diff
@@ -0,0 +1,61 @@
+import http from 'node:http'
+import events from 'node:events'
+import express from 'express'
+import { TestPlc } from './plc'
+import { TestPds } from './pds'
+import { TestBsky } from './bsky'
+import { TestOzone } from './ozone'
+
+export class IntrospectServer {
+  constructor(
+    public port: number,
+    public server: http.Server,
+  ) {}
+
+  static async start(
+    port: number,
+    plc: TestPlc,
+    pds: TestPds,
+    bsky: TestBsky,
+    ozone: TestOzone,
+  ) {
+    const app = express()
+    app.get('/', (_req, res) => {
+      res.status(200).send({
+        plc: {
+          url: plc.url,
+        },
+        pds: {
+          url: pds.url,
+          did: pds.ctx.cfg.service.did,
+        },
+        bsky: {
+          url: bsky.url,
+          did: bsky.ctx.cfg.serverDid,
+        },
+        ozone: {
+          url: ozone.url,
+          did: ozone.ctx.cfg.service.did,
+        },
+        db: {
+          url: ozone.ctx.cfg.db.postgresUrl,
+        },
+      })
+    })
+    const server = app.listen(port)
+    await events.once(server, 'listening')
+    return new IntrospectServer(port, server)
+  }
+
+  async close() {
+    return new Promise<void>((resolve, reject) => {
+      this.server.close((err) => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve()
+        }
+      })
+    })
+  }
```

Comment:
Since we have the `await events.once(server, 'listening')` line above—another way to slice this is:
```suggestion
    this.server.close()
    await events.once(this.server, 'close')
```

Reply:
ah nice I dig 👍